salt/cni/update-pre-orchestration.sls

```diff
@@ -11,7 +11,7 @@ include:
     - source: salt://cni/update-pre-orchestration.sh
     - mode: 0755
   cmd.run:
-    - name: /tmp/update-pre-orchestration.sh {{ grains['caasp_fqdn'] }} {{ salt.caasp_net.get_primary_ip() }} {{ salt.caasp_pillar.get('flannel:backend', 'vxlan') }}
+    - name: /tmp/update-pre-orchestration.sh {{ grains['nodename'] }} {{ salt.caasp_net.get_primary_ip() }} {{ salt.caasp_pillar.get('flannel:backend', 'vxlan') }}
```

Comment:
This is my only concern. This script will be called pretty early on the orchestration when we are migrating from non-cni to cni deployments (2.0 -> 3.0). This script uses the node name to patch the nodes and perform other operations. If this script is executed before we really rename the nodes, those patches to cni deployment will not succeed, as the node names will not exist (they haven't been renamed yet).

Reply:
Oh, humm - right. This may lead to the CNI upgrade code not working right at all. I'll look into this.

Reply:
What if we first restart the kubelet with the new `nodename` and then we run this code?

Reply:
One of the reasons of why it was moved as a first operation before we orchestrate was because the non-cni -> cni migration wasn't working properly: https://github.com/kubic-project/salt/pull/201#issuecomment-353984591. Specifically:

```
Looks like something during the orchestration makes them request a new podCIDR before we even set it with the orchestration mechanisms in place.
```

If we first restart the kubelet it will request a new `podCIDR`, we patch afterwards, but the `podCIDR` requested will still be assigned.